cluster/cluster.go

```diff
@@ -107,7 +107,7 @@ func Join(
 		readyc: make(chan struct{}),
 		logger: l,
 	}
-	p.delegate = newDelegate(l, reg, p)
+	p.delegate = newDelegate(l, reg, p, len(knownPeers)/2)
```

Comment:
nit but I would do the computation here rather than add a new argument to `newDelegate()`.

Reply:
you mean setting `RetransmitMulti` on p.delegate?

Reply:
Sorry for being inaccurate. I meant:

```go
retransmit := len(knownPeers)/2
if restransmit < 3 {
    retransmit = 3
}
p.delegate = newDelegate(l, reg, p, retransmit)
```